src/azure-cli/azure/cli/command_modules/synapse/manual/operations/workspace.py

```diff
@@ -124,27 +124,48 @@ def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin
         encryption = EncryptionDetails(cmk=CustomerManagedKeyDetails(key=workspace_key_detail, kek_identity=kek_identity))
     if user_assigned_identity_id:
         if user_assigned_identity_action == 'Add':
-            existing_identity = existing_ws.identity
-            keysList = list(existing_identity.user_assigned_identities.keys())
+            if existing_ws.identity.user_assigned_identities:
+                for uami_id in user_assigned_identity_id:
+                    userAssignedIdentities = UserAssignedManagedIdentity()
+                    existing_ws.identity.user_assigned_identities[uami_id] = userAssignedIdentities
+                identity = existing_ws.identity
+            else:
+                keysList = []
+                for uami_id in user_assigned_identity_id:
+                    keysList.append(uami_id)
+                userAssignedIdentities = UserAssignedManagedIdentity()
```

Comment:
This logic can be simplified, if user_assigned_identities is null, then new a dictionary, then all the other add uami can reuse the same code.